sdk/metric/view/view.go

```diff
@@ -17,22 +17,197 @@
+// will be used together to determine match and transforms.
 type Option interface {
 	apply(Config) Config
 }
 
-// TODO (#2837): implement view match options.
-// TODO (#2837): implement view annotation options.
+type optionFunc func(Config) Config
+
+func (f optionFunc) apply(cfg Config) Config {
+	return f(cfg)
+}
+
+// MatchInstrumentName will do an exact match of the name of the instrument.
+// Not compatible with MatchInstrumentNameRegexp.
+func MatchInstrumentName(name string) Option {
+	return optionFunc(func(cfg Config) Config {
+		cfg.instrumentName = name
+		return cfg
+	})
+}
+
+// MatchInstrumentNameRegexp will match any instrument with the provided
+// regexp.
+// Not compatible with MatchInstrumentName or WithName.
+func MatchInstrumentNameRegexp(re *regexp.Regexp) Option {
+	return optionFunc(func(cfg Config) Config {
+		cfg.instrumentNameRegexp = re
+		return cfg
+	})
+}
+
+// TODO: Implement when InstrumentKind and NumberKind are defined
+// func MatchInstrumentKind(k sdkinstrument.Kind) Option {
+// 	return optionFunc(func(cfg Config) Config {
+// 		cfg.instrumentKind = k
+// 		return cfg
+// 	})
+// }
 
-// New returns a new and configured view Config.
-func New(opts ...Option) Config {
-	return Config{}
+// func MatchNumberKind(k number.Kind) Option {
+// 	return optionFunc(func(cfg Config) Config {
+// 		cfg.numberKind = k
+// 		return cfg
+// 	})
+// }
+
+// MatchInstrumentationLibrary will do an exact match on any
+// instrumentation.Library field that is not blank ("").
+func MatchInstrumentationLibrary(lib instrumentation.Library) Option {
+	return optionFunc(func(cfg Config) Config {
+		cfg.library = lib
+		return cfg
+	})
+}
+
+// WithName will rename the instrument the view matches. If not used or empty the
+// instrument name will not be changed. Not compatible with
+// MatchInstrumentNameRegexp.
+func WithName(name string) Option {
+	return optionFunc(func(cfg Config) Config {
+		cfg.viewName = name
+		return cfg
+	})
+}
+
+// WithDescription will change the description of the instruments the view
+// matches. If not used or empty the description will not be changed.
+func WithDescription(desc string) Option {
+	return optionFunc(func(cfg Config) Config {
+		cfg.description = desc
+		return cfg
+	})
 }
+
+// WithKeyFilter will select attributes that have a matching key.  If not used
+// or empty no filter will be applied.
+func WithKeyFilter(keys ...attribute.Key) Option {
+	return optionFunc(func(cfg Config) Config {
+		if cfg.filterKeys == nil {
+			cfg.filterKeys = map[attribute.Key]struct{}{}
+		}
+		for _, key := range keys {
+			cfg.filterKeys[key] = struct{}{}
+		}
+		return cfg
+	})
+}
+
+// TODO: Implement when Aggregations and Temporalities are defined
+// func WithAggregation(kind aggregation.Kind) Option {
+// 	return optionFunc(func(cfg Config) Config {
+// 		cfg.aggregation = kind
+// 		return cfg
+// 	})
+// }
+
+// func WithTemporality(tempo aggregation.Temporality) Option {
```

Comment:
BTW, my read of the spec doesn't allow Views to control temporality. It's a per-instrument-kind selection associated with the reader.